src/api/llm/http.ts

```diff
@@ -72,3 +72,28 @@ export async function sendUserRating(
     throw new Error('Unexpected status code: ' + resp.status);
   }
 }
+
+export async function fetchAppNames(creds: Credentials) {
+  const response = await fetch(
+    `https://${getDomain(creds.region)}/v1/storage/files?kind=ios&kind=android`,
+    {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: 'Basic ' + btoa(`${creds.username}:${creds.accessKey}`),
+      },
+    },
+  );
+
+  if (!response.ok) {
+    throw new Error('Network response was not ok');
+  }
+
+  const data = (await response.json()) as { items: AppInfo[] };
+
+  if (data && Array.isArray(data.items) && data.items.every(isAppInfo)) {
```

Comment:
This is a strange mix of being defensive about the response data while making assumptions about it. Why assume the response has an `items` property but then check its items are the correct type? I'd think you'd want to verify the entire shape of the response if you're going to at all. And then have a single type guard instead of checking `isAppInfo` on all the items.